Monster: Cleanup/Refactor http controllers.

mnemonicly opened the pull request:
This cleans up docblocks, pulls most non-crudy actions into their own
controllers, and does general cleanup/logic refactoring.  There /should/
be no functional changes, but we all know how should works..

I'm happy to split this up into multiple smaller PR's if that is going to be cleaner to review.

mnemonicly committed:
Extract a handlesimages trait to centralize logic for parsing/storing images on upload in create/edit methods.

mnemonicly committed:
Use same image upload/layout in accessories as consum+components.

mnemonicly committed:
Monster: Cleanup/Refactor http controllers.

This cleans up docblocks, pulls most non-crudy actions into their own
controllers, and does general cleanup/logic refactoring.  There /should/
be no functional changes, but we all know how should works..

Extract checkin/checkout functions to a separate controller for accessories.

Move controllers to subdirectory.

Cleanup AssetModelsController

Extract component checkin/checkout

Assorted cleanups/doc/formatting in controllers.

Refactor LicenseController.

Refactor UsersController

Update viewassetscontroller.

mnemonicly committed:
Codacy cleanups

mnemonicly committed:
More codacy cleanups.  Extract a LicenseCheckout Form request as well.

mnemonicly committed:
A bit more refactor/cleaning of the license checkout method.

uberbrady reviewed (commented):
I'm feeling kinda ambivalent, maybe because I have something different in my head for how to clean some of this stuff up.

But I *do* very much like the fact that each checkin/checkout controller has a `create` and then a `store` method - that makes it certainly start to _feel_ a little more like they deserve their own controllers for these things.

My initial instinct is that checkin and checkout should just be one grand unified controller but then you do certainly lose the kinda REST-ish `create` and `store` methods that are cleanly one-per-file.

I originally came to Laravel from a Rails background, so I'm used to the 'fat model, skinny controller' paradigm. So my thought for how I wanted to see this cleaned up was this: I wanted more of the checkin/checkout logic that applies to the specific things that can be checked out or in to be embedded in the various things themselves - and then have the controller methods be much thinner, containing less asset-specific or accessory-specific or component-specific logic in it. If we could keep the 'smarts' embedded well enough in the model.

This could - eventually - get us to a "Checkoutable" trait that we can apply to all things that can be checked out, which would make it easier for us to really start to get rid of more duplicated code.

It *might* be possible to create a generic checkin/checkout controller that just calls `$thing->checkout($recipient)` or `$thing->checkin($optional_checkin_to_where)` - and then be able to just re-apply that generic controller across all the checkin/checkout types.

*That* would definitely be something I could get behind. We'd probably have to use some kind of polymorphism to figure out which view to show, but, still - that seems like it'd get rid of a lot of code, and could really add some value here.

That being said, when I first saw this I didn't like it - I do like my things to be in one file so I don't have, like, 15 files open when I'm trying to troubleshoot something, for example. But after having looked closer I don't hate it anywhere near as much. I could really see it going either way.

mnemonicly committed:
Review Related Cleanups

* Fix most of the item_not_found translations.  In many cases, the
string being generated did not even use the id parameter.  Where it
does, pass it as id instead of as a different value.

* Remove some old $data arrays from when we manually sent emails from
the controllers.  This has been superseeded by the notification system
(yay!)

mnemonicly committed:
Bugfix: Only log the checkin of an accessory if the checkin completes sucessfully.
